saleor/checkout/utils.py

```diff
@@ -178,26 +177,30 @@ def add_variants_to_checkout(
     to_create = []
     to_update = []
     to_delete = []
-    for variant, quantity, custom_price in zip(variants, quantities, custom_prices):
+    for variant, line_data in zip(variants, checkout_lines_data):
         if variant.pk in variant_ids_in_lines:
             line = variant_ids_in_lines[variant.pk]
-            if quantity > 0:
-                if replace:
-                    line.quantity = quantity
+            if line_data.quantity_to_update:
```

Comment:
[Question]. There is a lot of intendations here, maybe we could add some helper methods like:
```
for variant, line_data in zip(variants, checkout_lines_data):
    _append_lines_to_create(to_create,...)
    _append_lines_to_update(to_update, ...)
    _append_lines_to_delete(to_delete, ...)
 ```

Some of the below conditions probably will be duplicated but I think that this would be much more cleaner to read.

Reply:
I cannot separate this like that, but I can try to refactor it a little.